scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+        files.append(F[0]['full_path_on_store'])
+        sizes.append(F[1]['size'])
+        obsids.append(F[2]['obsid'])
+        start_jds.append(F[2]['start_time_jd'])
+    logger.info(f'Backing up JDS {"".join([str(jd) for jd in jds_to_backup])}. Nfiles = {len(files)}, Total Size = {np.round(np.sum(sizes)/1e12,1)}TB')
+    def update_tape_usage(tapes,testing=False):
+        #input: dict  from query_tape_usage and a session for heratape db
+        #output: input dict with added key: usage (sum in bytes known to heratape for each tape)
+        for i in range(len(tapes)):
+            tapeusage = query_tape_usage(tapes[i]['tape_id'],testing=testing)
+            tapes[i]['usage'] =  tapeusage
+        return tapes
+    def select_empty_tape(drive_id,testing=False):
+        # Impliments tape selection logic
+        # drive 1 uses tape slots 1-12, drive 2 uses 13-24
+        # there needs to be one empty slot for shuffling and another is taken up by a cleaner tape
+        # ideally these are split between the two halves of the jukebox
+        # input: drive number 0 or 1
+        # get the next tape in the relevant half which is not full
+        # full being defined as > 50%
+        # return: slot, tape_id, and tape_usage (in bytes) to be used
+        TAPECAPACITY = 18e12 #TODO: replace this with the number in heratape.Tapes
+        assert(drive_id  in [0,1])
+        tape_archive = query_tape_jukebox() #get the list of tapes currently in the jukebox
+        tape_archive = update_tape_usage(tape_archive,testing=testing) #find out how full they are.
+        #throw out any that are too full
+        empty_tapes = []
+        for tape in tape_archive:
+            if tape['tape_id'].startswith('CLN'):continue#ignore the cleaner!
+            logger.info(tape['usage'])
+            if tape['usage'] < TAPECAPACITY *0.5:
+                logger.info(tape['tape_id'])
+                if (int(tape['slot']) <= 12) == (drive_id==0): # slot<=12 and drive 0, or both opposite.
+                    empty_tapes.append(tape)
+        # a nice flourish, lets use the lowest serial number first
+        usetape_id = sorted([t['tape_id'] for t in empty_tapes])[0]
+        for tape in empty_tapes:
+            if tape['tape_id'] == usetape_id:
+                return tape['slot'], tape['tape_id'], tape['usage'] 
+        return None, None, None # if no tapes found, its probably time for a fresh batch of tapes!
+        
+    #Armed with all our metadata from the Librarian, we can move to the Tape side of things
+    
+    # prepare tape for backup
+    drivetapeid = tape_in_drive(mydrive)
+    logger.info(f'Drive {mydrive}, Tape {drivetapeid}')
+    #check the total against the remaining space
+    tape_usage = query_tape_usage(drivetapeid,testing=TESTING)
+    if tape_usage >0:
+        logger.info(f'Size of this backup = {np.sum(sizes)/1e12} TB is < the {(TAPESIZE - tape_usage)/1e12}TB theoretically remaining on the tape')
```

Comment:
I don't think that the calculation in this logging message is actually happening. The actual test is whether the tape is more or less than half full. So this message is misrepresenting the logic in the code.